src/Umbraco.Infrastructure/Persistence/Repositories/Implement/MediaRepository.cs

```diff
@@ -648,6 +706,33 @@ protected override Sql<ISqlContext> GetBaseQuery(bool isCount) =>
 
         protected override string GetBaseWhereClause() =>
             throw new InvalidOperationException("This method won't be implemented.");
+
+        /// <summary>
+        /// Populates the GUID-keyed cache with the given entity.
+        /// This allows entities retrieved by int ID to also be cached for GUID lookups.
+        /// </summary>
+        public void PopulateCache(IMedia entity)
```

Comment:
I would really like all these to be named `PopulateXCache` or something similar. Just because they have the exact same name everywhere, but some populate Id, some populate key 🤔 
I know there are also XML comments and inline comments, but still 🙈